src/main/java/com/google/step/YOUR_PROJECT_NAME_HERE/external/W3SchoolClient.java

```diff
@@ -0,0 +1,42 @@
+package com.google.step.YOUR_PROJECT_NAME_HERE.external;
+
+import com.google.step.YOUR_PROJECT_NAME_HERE.data.Card;
+import java.io.IOException;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.select.Elements;
+
+public final class W3SchoolClient {
+  private static final String CARD_TITLE_ID = "h1";
+  private static final String CARD_DESC_ID = "p";
+  private static final String CARD_SNIPPET_ID = "w3-example";
+  private static final String CARD_CODE_ID = "w3-code";
+
+  public Card search(String w3Link) throws IOException {
+
+    Document doc = Jsoup.connect(w3Link).get();
```

Comment:
Wow, that's the most succinct GET request I've seen in Java ever.

Reply:
But you could probably make it even more concise with Jsoup.parse? Only necessary if you all end up making the interface to the parsers a URI instead of a String.